GameVersion.cpp

```diff
@@ -154,19 +276,89 @@ GameVersion GameVersion::Factory::getVersion(std::istream& theStream)
 
 std::string GameVersion::toString() const
 {
-	return std::to_string(firstPart) + "." + std::to_string(secondPart) + "." + std::to_string(thirdPart) + "." +
-			 std::to_string(fourthPart);
+	std::string toReturn;
+	if (firstPart)
+		toReturn += std::to_string(*firstPart) + ".";
+	else
+		toReturn += "0.";
+	if (secondPart)
+		toReturn += std::to_string(*secondPart) + ".";
+	else
+		toReturn += "0.";
+	if (thirdPart)
+		toReturn += std::to_string(*thirdPart) + ".";
+	else
+		toReturn += "0.";
+	if (fourthPart)
+		toReturn += std::to_string(*fourthPart);
+	else
+		toReturn += "0";
+
+	return toReturn;
 }
 
 std::string GameVersion::toShortString() const
 {
 	std::string nameString;
-	if (fourthPart > 0)
-		nameString = "." + std::to_string(fourthPart);
-	if (thirdPart > 0 || !nameString.empty())
-		nameString = "." + std::to_string(thirdPart) + nameString;
-	if (secondPart > 0 || !nameString.empty())
-		nameString = "." + std::to_string(secondPart) + nameString;
-	nameString = std::to_string(firstPart) + nameString;
+	if (fourthPart && *fourthPart > 0)
+		nameString = "." + std::to_string(*fourthPart);
+
+	if (thirdPart)
+	{
+		if (*thirdPart > 0)
+			nameString = "." + std::to_string(*thirdPart) + nameString;
+		else if (!nameString.empty())
+			nameString = ".0" + nameString;
+	}
+
+	if (secondPart)
+	{
+		if (*secondPart > 0)
+			nameString = "." + std::to_string(*secondPart) + nameString;
+		else if (!nameString.empty())
+			nameString = ".0" + nameString;
+	}
+
+	if (firstPart)
+	{
+		if (*firstPart > 0)
+			nameString = std::to_string(*firstPart) + nameString;
+		else
+			nameString = "0" + nameString;
+	}
+
 	return nameString;
 }
+
+bool GameVersion::isLargerishThan(const GameVersion& rhs) const
+{
+	// Largerish is intended for fuzzy comparisons like "converter works with up to 1.9",
```

Comment:
Please also put this comment in the header so people don't have to check the .cpp file to know what this function does.